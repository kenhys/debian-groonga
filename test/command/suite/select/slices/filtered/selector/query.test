table_create Users TABLE_NO_KEY
column_create Users name COLUMN_SCALAR ShortText

table_create Lexicon TABLE_HASH_KEY ShortText \
  --default_tokenizer TokenBigramSplitSymbolAlphaDigit \
  --normalizer NormalizerAuto
column_create Lexicon users_name COLUMN_INDEX|WITH_POSITION Users name

load --table Users
[
{"name": "Alice"},
{"name": "Alisa"},
{"name": "Bob"}
]

log_level --level info
#@add-important-log-levels info
select Users \
  --filter '_id > 1' \
  --slices[name].filter 'query("name * 10", "ali")' \
  --slices[name].output_columns _id,name,_score
