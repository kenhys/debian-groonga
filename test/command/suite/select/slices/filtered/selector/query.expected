table_create Users TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Users name COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Lexicon TABLE_HASH_KEY ShortText   --default_tokenizer TokenBigramSplitSymbolAlphaDigit   --normalizer NormalizerAuto
[[0,0.0,0.0],true]
column_create Lexicon users_name COLUMN_INDEX|WITH_POSITION Users name
[[0,0.0,0.0],true]
load --table Users
[
{"name": "Alice"},
{"name": "Alisa"},
{"name": "Bob"}
]
[[0,0.0,0.0],3]
log_level --level info
[[0,0.0,0.0],true]
select Users   --filter '_id > 1'   --slices[name].filter 'query("name * 10", "ali")'   --slices[name].output_columns _id,name,_score
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "name",
          "ShortText"
        ]
      ],
      [
        2,
        "Alisa"
      ],
      [
        3,
        "Bob"
      ]
    ],
    {
      "name": [
        [
          1
        ],
        [
          [
            "_id",
            "UInt32"
          ],
          [
            "name",
            "ShortText"
          ],
          [
            "_score",
            "Int32"
          ]
        ],
        [
          2,
          "Alisa",
          11
        ]
      ]
    }
  ]
]
#|i| [table][select][index][selector][no-index][query] <>
#|i| [object][search][index][key][exact] <Lexicon.users_name>
#|i| grn_ii_sel > (ali)
#|i| n=2 (ali)
#|i| exact: 2
#|i| hits=2
#|i| [accessor][resolve][table] (temporary) -> <Users>
